service/history/replication/task_ack_manager.go

```diff
@@ -191,9 +225,12 @@ func (t *TaskAckManager) getTasks(ctx context.Context, pollingCluster string, la
 		return nil, err
 	}
 
-	t.scope.RecordTimer(metrics.ReplicationTasksLag, time.Duration(t.ackLevels.UpdateIfNeededAndGetQueueMaxReadLevel(persistence.HistoryTaskCategoryReplication, pollingCluster).GetTaskID()-msgs.LastRetrievedMessageID))
-	t.scope.RecordTimer(metrics.ReplicationTasksReturned, time.Duration(len(msgs.ReplicationTasks)))
-	t.scope.RecordTimer(metrics.ReplicationTasksReturnedDiff, time.Duration(len(taskInfos)-len(msgs.ReplicationTasks)))
+	t.metrics.replicationLag(
+		t.ackLevels.UpdateIfNeededAndGetQueueMaxReadLevel(persistence.HistoryTaskCategoryReplication, pollingCluster),
+		taskInfos,
+		msgs,
+		t.scope,
+	)
```

Comment:
I've done it this way _mostly_ because:
1. the metrics-func is private and local to this file, so it's not too concerning to have it depend on excessively-large objects
2. each argument is a unique type, so it's not possible to pass them wrong (the main alternative is to pass three integers).